packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -4431,6 +4441,72 @@ const tests = {
           `This lets you handle multiple requests without bugs.`,
       ],
     },
+    {
+      code: `
+        function Thing() {
+          const ref = React.createRef();
+          useEffect(() => {
+            console.log(ref.current);
+          }, [ref]);
+        }
+      `,
+      errors: [
+        `React Hook useEffect has a dependency that was created by ` +
+          `creatRef(). This will create a new ref object on every re-render ` +
+          `so that the hook will always need to re-run. Did you mean to use ` +
+          `useRef() instead?`,
+      ],
+    },
+    {
+      code: `
+        function Thing() {
+          const ref = React.createRef();
+          useEffect(() => {
+            console.log(ref.current);
+          }, []);
+        }
+      `,
+      errors: [
+        `React Hook useEffect has a dependency that was created by ` +
+          `creatRef(). This will create a new ref object on every re-render ` +
+          `so that the hook will always need to re-run. Did you mean to use ` +
+          `useRef() instead?`,
```

Comment:
I think in this case we need a different message as it _does not_ have a dependency and won't re-run on every render. I'd say we still encourage this pattern even if the function captures only the first ref value and not the ones on re-render.